src/Application.cpp

```diff
@@ -55,7 +56,24 @@ void Application::render () {
 }
 
 void Application::update () {
-	scenes[currentScene]->OnUpdate();
+	scenes[currentScene]->OnUpdate(elapsedTime);
+}
+
+void Application::changeSceneTo(int sceneIndex){
+	if(sceneIndex < 0 || sceneIndex >= scenes.size()) {
+		std::cout<<"scene index non-existend\n";
+		return;
+	}
+	currentScene = sceneIndex;
+}
+void Application::pushBackScene(Scene& scene){
+	scenes.push_back(&scene);
+}
+void Application::removeScene(int sceneIndex){
+	scenes.erase(scenes.begin()+sceneIndex);
+}
```

Comment:
As I mentioned, using a `sceneIndex` to change or remove scenes could result in weird behavior that would be difficult to debug. An example of this being that if we were to remove a scene, the scene index for all scenes placed on the stack after that scene will decrease by one.